streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java

```diff
@@ -781,4 +782,17 @@ public void shouldNotAllowToAddGlobalStoreWithSourceNameEqualsProcessorName() {
             sameNameForSourceAndProcessor,
             new MockProcessorSupplier());
     }
+
+    @Test
+    public void shouldAllowAddGlobalStoreWithNonKVStore() {
+        final StoreBuilder windowStoreBuilder = new MockWindowStoreBuilder("store", false);
+        builder.addGlobalStore(windowStoreBuilder.withLoggingDisabled(),
```

Comment:
The signature of `addGlobalStore` accepts any `StoreBuilder`, thus I am wondering if we should switch to pass in a `TestStateStoreBuilder` that returns `StateStore` instead of a `WindowedStore`?

I am also thinking if we should add an test using `TopologyTestDriver` to verify that a `TestStateStore` can be used as global store?